### DISCUSSION TITLE
{"context": "core"} {"context": "core/relayer"} [Error: socket stalled]

### DISCUSSION THREAD

**Discussion by yashwanthmuddana** [WalletConnect/walletconnect-monorepo#2793] [Developer Support] [answered]
Error: Attempt to connect to relay via `transportOpen` has stalled. Retrying...

getting this sometimes!, could i know why its happening

using React native version: 0.71.11

using : @walletconnect/modal-react-native

relayUrl: wss://relay.walletconnect.org

**Comment by crypblizz8** [comment]
Hey there, please provide more information on your 
- Reproducible code via a codeBox / codeblock here


CC @ignaciosantise

**Comment by siosio34** [comment]
이 에러는 나또한 발생한다.

**Comment by ignaciosantise** [comment]
@yashwanthmuddana are you using a VPN? If so, can you try disabling it to see if it works?

**Reply by agam-w** [reply]
I also facing this issue using "@walletconnect/modal": "2.4.7", im using **nextjs**. if i turn off my VPN it will works like @ignaciosantise suggest, but can we handle this issue from ourside? so we still can connect even using VPN? or walletconnect team will fix this issue soon? 

really appreciate for your help @ignaciosantise , thanks.

**Reply by ignaciosantise** [reply, marked as answer]
@agam-w the root problem is that some VPN/ISP block `.com` urls, and our default relay url is `wss://relay.walletconnect.com`

We've just released `"@walletconnect/modal": "2.5.9"` that uses `.org` in case `.com` is blocked. Can you check if that update works for your case?

**Reply by yashwanthmuddana** [reply]
Thank you for the help @ignaciosantise. but dint used any vpn back then, after that dint came accross that error again after adding 
`relayUrl:"wss://relay.walletconnect.org` in WalletConnetModal from "@walletconnect/modal-react-native"

**Reply by anegg0** [reply]
Thank you @ignaciosantise, in my case, the VPN was the cause of the stalled socket.

**Comment by ehsan-g** [comment]
Blocking Iranian IP addresses was not a very Decenterlized move...

**Comment by ligi** [comment]
@ehsan-g can you provide context? Did WalletConnect block iranian IP's ?

**Comment by ehsan-g** [comment]
@ligi  I am using WC for my open source project. The walletConnect.org relay is only available when I use vpn, I am using Wagmi and Viem to use my providers and here is my setup and the errors:
Code: https://github.com/SAY-DAO/front-end-v2/blob/release/src/wallet.js
Error:
1- `Firefox can’t establish a connection to the server at wss://relay.walletconnect.org/?auth=...`
2- `Object { context: "core/relayer" } Error: WebSocket connection failed for host: wss://relay.walletconnect.org`
3- `Object { context: "client" } WebSocket connection failed for host: wss://relay.walletconnect.org`
4- `Uncaught (in promise) Error: WebSocket connection failed for host: wss://relay.walletconnect.org`
5- `GET =>wss://relay.walletconnect.org/?auth => 403 Forbidden`

To double check I cloned the Uniswap/web3-react and  I tried with no Vpn, it returned : Error: WebSocket connection failed for host: wss://relay.walletconnect.org
